src/core/storage/client.ts

```diff
@@ -144,8 +188,7 @@ export class StorachaClient implements StorageClient {
       if ((error instanceof Error && error.name === 'AbortError') || options.signal?.aborted) {
         throw new Error('Upload aborted');
       }
-      const message = error instanceof Error ? error.message : 'Unknown error';
-      throw new Error(`Upload failed: ${message}`);
+      throw error;
```

Comment:
I'm not sure this catch block is adding any value. In fact, you're actually making it harder to distinguish between an abort vs another error by transforming the abort error into a regular error.

Reply:
Good point. I will just log the error and throw it again. thanks